### DISCUSSION TITLE
Can I fetch all txn info related with nft(list, buy, create collection, mint, burn, transfer) on Aptos chain?

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#512] [Questions]
### Discord user ID

aptain7

### Describe your question in detail.

How can I monitor all transactions and fetch the events that mentioned?
Not only specific contract, but whole mainnet.
If catch one, then how to fetch info(sender, buyer etc)

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by ying-w** [comment]
For all of mainnet you'll need to run your own indexer or use free tier for grpc endpoint

See https://aptos.dev/en/build/indexer/indexer-reference#token_activities_v2

While collection (create), digital asset (mint,burn), object (transfer) are primitives provided by aptos, listing in marketplace is implemented custom and will prob have a non-standardized format
